IMP-2284 Remove EDS and other legacy code from Bento

jazairi opened the pull request:
#### Why these changes are being introduced:

With the launch of Alma and Primo, we no longer need the EDS code
in Bento, and removing it now with save us a major headache in the
future. It also make sense at this point to remove other features
we no longer use, like the button makers, Aleph RTA, etc.

#### Relevant ticket(s):

https://mitlibraries.atlassian.net/browse/IMP-2284

#### How this addresses that need:

This removes several features, including:

* All EDS logic (normalization and search models, as well as code
from the search and record controllers and views, and result model
and views);
* Full records, with the exception of some logic required for Primo
redirects;
* Aleph real-time availability;
* SFX handler;
* Button makers;
* Rainbows; and
* Authentication.

#### Side effects of this change:

* Unused VCR cassettes have been removed.
* The system diagram has been updated.
* Some irrelevant assets have been removed (Aleph RTA JS and SCSS
for removed features).

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

JPrevost commented on app/views/search/search_boxed.html.erb:
```diff
@@ -57,7 +51,6 @@
 
   <script>
-    RealtimeStatus( '#<%= id %>' );
     TrackLinks( $( '#<%= id %>' ).find( 'a' ) );
```
Noted elsewhere but I don't think we use TrackLinks anymore. Likely worth a dedicated ticket to come back and clean it up.

JPrevost commented on config/features.rb:
```diff
@@ -7,46 +7,6 @@
     default: false,
```
is debug still used anywhere? I think I saw the partial pulled out but I suppose it may be used elsewhere...

JPrevost commented on config/features.rb:
```diff
@@ -7,46 +7,6 @@
-    default: ENV['DISABLE_WORLDCATINATE'],
-    description: 'Determines if links for renovation items should be enabled'
-
   feature :primo_search,
```
I'm curious if both `primo_search` and `primo_redirects` should be removed as features and left as on. I'm happy to hear thoughts on why to keep them though.

jazairi replied:
Resolving this one since we have an open discussion in another thread.

JPrevost reviewed (commented):
Take a look through the comments I've made and let me know if there are any you'd like to address now, open tickets on, or disagree with.

Regardless, before merging let's ask UX to sign off on the functionality in the PR build.

Looks good!

jazairi commented on app/controllers/record_controller.rb:
```diff
@@ -1,155 +1,15 @@
+  # We need to leave this method and direct_link because the record links have historically been used as permalinks and 
+  # should therefore be redirected to Primo records indefinitely.
   def record
     if Flipflop.enabled?(:primo_redirects)
```
Good point. That seems reasonable to me.

jazairi commented on app/controllers/search_controller.rb:
```diff
@@ -108,16 +89,6 @@ def validate_q!
-  def paginate_results(page, per_page)
-    # EDS API Limits retrieving results past about 250
-    max = [@results['total'], 250].min
-    Kaminari.paginate_array(
```
I suspect it is... Will confirm and remove if so.

jazairi commented on app/models/result.rb:
```diff
@@ -20,96 +16,8 @@ def initialize(title, url)
-      an.split('.').last.prepend('MIT30')
-    elsif alma_record?
-      an
+    if Flipflop.enabled? :primo_search
```
Matt and I discussed it while you were on vacation. The rationale for keeping it is essentially to ease migrations to future discovery backends. I.e., when we migrate from Primo to Catalogtron 4000, we will already have the Primo feature flag in place that we can switch off when we're ready to launch. It's an unlikely scenario that we'll still be using this Bento app at that point, but it seemed relatively harmless to me to leave the feature flag.

This is not a strongly held opinion, and I'm happy to remove the flag if you think it's more confusing than potentially useful. I should note that this and debug mode are the last two feature flags, which prompts the question of whether we should remove Flipflop altogether.

jazairi replied:
Thinking about this further, it might not make a lot of sense to have a feature flag that, when disabled, functionally breaks the app. It's probably more reasonable to introduce a new feature flag for whatever the next catalog is (assuming this app lasts that long).

I would say, though, that along with removing the feature flag, we should probably also remove Flipflop. Does that make sense?

JPrevost replied:
Yeah let's leave it in for now and reconsider in a few months as we start to get a better feel for how our Discovery Roadmap is looking.

jazairi commented on app/views/search/_result.html.erb:
```diff
@@ -94,50 +85,13 @@
-        <%= link_to("Check for online access", result.check_sfx_url, class: 'online button-secondary', data: {type: "Check"}) %>
-      <% end %>
-    <% end %>
-
```
Good catch. I think that also renders the feature flag irrelevant, since we don't use it anywhere else in the app.

jazairi replied:
Good call, I'll remove that as well as the feature flag.

jazairi commented on app/views/search/_trigger_search.html.erb:
```diff
@@ -5,7 +5,6 @@ $.ajax({
   $('#<%= id %>').html( msg );
   TrackLinks( $( this ).find( 'a' ) );
```
It might be clean enough that it's worth doing in this PR. If not, I'll open a new ticket.

jazairi commented on app/models/normalize_google.rb:
```diff
@@ -6,7 +6,7 @@ def to_result(results, q)
     norm['total'] = results.queries.request[0].total_results.to_i
     norm['results'] = []
-    norm['eds_ui_view_more'] = view_more(q)
+    norm['view_more'] = view_more(q)
```
Yeah. I _think_ that's the last of the hardcoded product names, or at least the last of it that affects anything noticeably...

jazairi commented:
@JPrevost Thanks for the thorough review! I've addressed most of the comments, but I left the primo_search feature flag until we discuss it further (see my response in the thread above). Still open to removing it -- even more so now that it's the only remaining feature flag -- but wanted to make sure we're considering all angles.

JPrevost reviewed (approved):
This is all good from the code side. Once you get sign off from Darcy/Melissa we can merge. 👍🏻

jazairi commented:
This now has stakeholder approval, but I have some questions about how we want to handle ENV requirements in app.json moving forward. We'll discuss that in Tuesday's EngX meeting, and this will merge once we have a decision and it's passed final code review.
